### PR TITLE
Baker updates

### PR DESCRIPTION
**Pull request by QuantumEntangledAndy**
This includes the updates discussed in #47 

---

### Features

- Switch to f32 based chunks so that isovalues can be used (and density)
- Add color and atlas to the marching cube baker
- Add modified marching cube baker that adds extra faces to the marching cube baker so that atlases are good
- Modify the voxel baker as discussed in #47 so that it works on a grid similar to marching cubes
- Added an rstar tree based deduplicating method
- Optimised the octree deduplicating method
- Added octree search
- Added a no deduplicating method
- Added an option to deduplicate the vertices during bake into baker options (default no deduplicating)
- Updated the amethyst terrain example to include marching and modified-marching cubes
- Added benchmarks for the deduplcating methods
- Added various tests
- Modifed the gox read to load a single chunk. This is neccecary because of how it chunks in a different way to how marching cubes (and now voxel) excepts

### CODE REVIEW

File: crates/gaiku_baker_modified_marching_cubes/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+[package]
+name = "gaiku_baker_modified_marching_cubes"
+version = "0.1.0"
+authors = ["Norman Paniagua <normanpaniagua@gmail.com>"]
```
Comment (norman784) [inline crates/gaiku_baker_modified_marching_cubes/Cargo.toml]: You can add yourself as the author too if you want

File: crates/gaiku_format_gox/src/lib.rs [hunk 2]
```diff
@@ -26,17 +28,86 @@ impl FileFormat for GoxReader {
...
+    let chunk_size: [Coord; 3] = [
+      (max[0] - min[0] + 1).try_into().unwrap(),
+      (max[1] - min[1] + 1).try_into().unwrap(),
+      (max[2] - min[2] + 1).try_into().unwrap(),
+    ];
+
+    let mut chunk = C::new(
+      [(min[0]) as f32, (min[2]) as f32, (min[1]) as f32],
+      chunk_size[0].try_into().unwrap(),
+      chunk_size[2].try_into().unwrap(), // goxel is in y up gaiku in zup
```
Comment (norman784) [inline crates/gaiku_format_gox/src/lib.rs]:
```suggestion
      chunk_size[2].try_into().unwrap(), // goxel is in y up gaiku in z up
```

**Review by norman784 — COMMENTED** [review]
I didn't testet it yet, but reading quickly the code seems all ok, later today I will try to build and run the tests locally. Also left a few comments

File: crates/gaiku_baker_modified_marching_cubes/Cargo.toml (hunk 1 above)
Comment (QuantumEntangledAndy) [reply crates/gaiku_baker_modified_marching_cubes/Cargo.toml]: Thanks, didn't want to presume since I am not too bothered either way

**Comment by norman784** [comment]
There's is a dependency in the amethyst integration that won't let the tests run, I could run the tests in common and the bakers without issues.

**Comment by QuantumEntangledAndy** [comment]
Do you mean the lexical_core one? It's been fixed upstream. You just needed to clean your target and delete the cargo.lock and it should work. That's what I had to do for it.

**Comment by norman784** [comment]
Ok, the tests passed locally but not here :thinking:, also clippy isn't working don't know if is a bug in clippy or something is off in our code.

**Comment by QuantumEntangledAndy** [comment]
Clippy for nightly is bugged upstream. I've looked into it and it's been fixed but it won't rollout until their next nightly sync in 8 days. 

The build failed on a crate download failure. Can probably just rerun that one.

**Review by norman784 — APPROVED** [review]
LGTM

**Pull request merged**